src/main/java/org/jabref/logic/cleanup/MoveFilesCleanup.java

```diff
@@ -16,64 +23,113 @@
 import org.jabref.model.entry.ParsedFileField;
 import org.jabref.model.metadata.FileDirectoryPreferences;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
 public class MoveFilesCleanup implements CleanupJob {
 
     private final BibDatabaseContext databaseContext;
     private final FileDirectoryPreferences fileDirectoryPreferences;
+    private final LayoutFormatterPreferences layoutPrefs;
+
+    private final String fileDirPattern;
+    private static final Log LOGGER = LogFactory.getLog(MoveFilesCleanup.class);
 
+    private ParsedFileField singleFileFieldCleanup;
 
-    public MoveFilesCleanup(BibDatabaseContext databaseContext, FileDirectoryPreferences fileDirectoryPreferences) {
+    public MoveFilesCleanup(BibDatabaseContext databaseContext, String fileDirPattern,
+            FileDirectoryPreferences fileDirectoryPreferences, LayoutFormatterPreferences layoutPrefs) {
         this.databaseContext = Objects.requireNonNull(databaseContext);
+        this.fileDirPattern = Objects.requireNonNull(fileDirPattern);
         this.fileDirectoryPreferences = Objects.requireNonNull(fileDirectoryPreferences);
+        this.layoutPrefs = Objects.requireNonNull(layoutPrefs);
+    }
+
+    public MoveFilesCleanup(BibDatabaseContext databaseContext, String fileDirPattern,
+            FileDirectoryPreferences fileDirectoryPreferences, LayoutFormatterPreferences prefs,
+            ParsedFileField field) {
+
+        this(databaseContext, fileDirPattern, fileDirectoryPreferences, prefs);
+        this.singleFileFieldCleanup = field;
     }
 
     @Override
     public List<FieldChange> cleanup(BibEntry entry) {
-        if(!databaseContext.getMetaData().getDefaultFileDirectory().isPresent()) {
+        Optional<Path> firstExistingFileDir = databaseContext.getFirstExistingFileDir(fileDirectoryPreferences);
```

Comment:
Sounds like a good idea. However, as my time is currently limited due to exams I would rather vote to merge it in as it is, as it fixes several important bugs and I would create a followup PR for making the code strucutred